src/tests/unit/tests/DetailsView/components/save-assessment-button.test.tsx

```diff
@@ -122,6 +110,14 @@ describe('SaveAssessmentButton', () => {
                     Times.atLeastOnce(),
                 );
             });
+            it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {
```

Comment:
Hi @madalynrose i have checked the issue for the test case **it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {**
so here i was able to reproduce the issue as per your comments
So i can see the 3rd render is happening on Primary button onClick (got it) on the Dialog. As mocking this button closes the Dialogue. Hence re-rendering happening to update the Dialogue state to hide.

Case-1: with onClick
it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {
                const gotItButtonProps = getMockComponentClassPropsForCall(PrimaryButton);
                act(() => {
                    gotItButtonProps.onClick();
                });
                const test = (Dialog as any).render.mock.calls;
                console.log('test--->', test)
                const getProps = getMockComponentClassPropsForCall(Dialog, 3);
                expect(getProps.hidden).toEqual(true);
            });
**Case -1 - Output:**:
![image](https://github.com/microsoft/accessibility-insights-web/assets/155614445/4a59716e-1303-4e46-ad7c-c3517ef03303)

**Case-2:without onClick**
it('dialog is hidden (dismissed) when "got it" button is clicked', async () => {
                const gotItButtonProps = getMockComponentClassPropsForCall(PrimaryButton);
                act(() => {
                    **//   gotItButtonProps.onClick();**
                });
                const test = (Dialog as any).render.mock.calls;
                console.log('test--->', test)
                const getProps = getMockComponentClassPropsForCall(Dialog, 3);
                expect(getProps.hidden).toEqual(true);
            });
**Case-2 - Output:**
![image](https://github.com/microsoft/accessibility-insights-web/assets/155614445/65a145f3-94b5-4660-8ee8-b308086c0330)


We can see in case2 output we can see only 2 renders.
Please let me know if it helps, or is my understanding of issue is correct?
Thanks

Reply:
Case 2 output still has 3 renders. The last one is `[]`, meaning that it was called without any props.

I suppose at this point we can change the test to look at the fourth render since we can prove that that's the one triggered by `onClick`. It'd be good to add a comment explaining why we're looking at the fourth and not the third though.

Reply:
Okay I think what we should do is just look at call 4 and add a comment about the weird behavior. I'll make a suggestion.